### PR TITLE
[INLONG-5242][Sort] Add metric computing for mongoDB and Oracle

### LINKED ISSUE #5242: [Feature][Sort] Add metric computing for Oracle and MongoDB
### Description

Add metric computing for mongodb and oracle

### Use case

User only need add with param` "inlong.metric":"<groupId>&<streamId>&<nodeId>"` when use flink sql.
example:
```sql
  create table `table_groupId_streamId_nodeId1`(
    `age` INT,
    `name` STRING,
    PRIMARY KEY(`age`) NOT ENFORCED)
    WITH (
    'connector' = 'mongodb-cdc-inlong',
   'hosts' = 'localhost:27017,localhost:27018,localhost:27019',
   'username' = 'flinkuser',
   'password' = 'flinkpw',
   'database' = 'inventory',
   'collection' = 'products'
   'inlong.metric' = 'pgGroup&pgStream&pgNode1'
);

CREATE TABLE `table_groupId_streamId_nodeId2`(
    PRIMARY KEY (`name`,`age`) NOT ENFORCED,
    `name` STRING,
    `age` INT)
    WITH (
    'connector' = 'print'
);

   INSERT INTO `table_groupId_streamId_nodeId2`
    SELECT
    `name` AS `name`,
    `age` AS `age`
    FROM `table_groupId_streamId_nodeId1`;
```
wo can set flink-conf.yaml add metric report
```xml
metric.reporters: promgateway
metrics.reporter.promgateway.class: org.apache.flink.metrics.prometheus.PrometheusPushGatewayReporter
metrics.reporter.promgateway.host: ip
metrics.reporter.promgateway.port: 9091
metrics.reporter.promgateway.interval: 60 SECONDS
metrics.reporter.promgateway.filterLabelValueCharacters: false
```

we can see metric in pushgateway.

I will add this feature for oracle cdc and mongodb cdc
[link issue](https://github.com/apache/inlong/issues/4967#event-6992075233)

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gong**
### Prepare a Pull Request

- [INLONG-5242][Sort] Add metric computing for mongodb and oracle
- Fixes #5242 

### Motivation

Add metric computing for mongodb and oracle

### Modifications

* Add metric computing for mongodb and oracle
* Add optional table option `inlong.metric` for mongodb and oracle
* Modify delimiter of `inlong.metric`, from  `_`  to  `&`. Because groupId and streamId can contain `_`, it will lead split error.
* Modify pom of postgres. Because it will cause lost dependency class about connector-base. 
* Improve connector-base, It will support custom Counter.

### CODE REVIEW

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public class PostgreSQLTableFactory implements DynamicTableSourceFactory {
             ConfigOptions.key("inlong.metric")
                     .stringType()
                     .defaultValue("")
-                    .withDescription("INLONG GROUP ID + '_' + STREAM ID + '_' + NODE ID");
+                    .withDescription("INLONG GROUP ID + '&' + STREAM ID + '&' + NODE ID");
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java:47]: It is recommended to use str.format.

Comment (gong) [reply inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java:47]: Here is description. not concat string

**Review by yunqingmoswu — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-connectors/oracle-cdc/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#
```
Comment (healchow) [inline inlong-sort/sort-connectors/oracle-cdc/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory:1]: Why two license headers?

**Review by GanfengTan — APPROVED** [review]

**Review by EMsnap — COMMENTED** [review]
Can DebeziumSourceFunction  be extracted?

File: inlong-sort/sort-connectors/oracle-cdc/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory (hunk 2 above)
Comment (gong) [reply inlong-sort/sort-connectors/oracle-cdc/src/main/resources/META-INF/services/org.apache.flink.table.factories.Factory:1]: remove duplication license header

**Review by dockerzhang — APPROVED** [review]

**Comment by gong** [comment]
> Can DebeziumSourceFunction be extracted?

some connector don't need dependency about debezium. So DebeziumSourceFunction can't be extracted.

**Review by healchow — APPROVED** [review]

**Pull request merged**